src/Network/Network/Models/AzureFirewallPolicy/PSAzureFirewallPolicy.cs

```diff
@@ -42,5 +47,77 @@ public class PSAzureFirewallPolicy : PSTopLevelResource
         public PSAzureFirewallPolicyTransportSecurity TransportSecurity { get; set; }
 
         public PSAzureFirewallPolicySku Sku { get; set; }
+
+        public PSAzureFirewallPolicySNAT Snat { get; set; }
+
+        private const string IANAPrivateRanges = "IANAPrivateRanges";
+
+        public string[] PrivateRange
+        {
+            get
+            {
+                return Snat?.PrivateRanges.ToArray();
+            }
+            set
+            {
+                if (value != null)
+                    ValidatePrivateRange(value);
+                Snat = new PSAzureFirewallPolicySNAT() { PrivateRanges = value };
+            }
+        }
+
+        [JsonIgnore]
+        public string PrivateRangeText
+        {
+            get { return JsonConvert.SerializeObject(PrivateRange, Formatting.Indented); }
+        }
+
+        #region Private Range Validation
+        private void ValidatePrivateRange(string[] privateRange)
+        {
+            foreach (var ip in privateRange)
+            {
+                if (ip.Equals(IANAPrivateRanges, StringComparison.OrdinalIgnoreCase))
+                    continue;
+
+                if (ip.Contains("/"))
+                    ValidateMaskedIpAddress(ip);
+                else
+                    ValidateSingleIpAddress(ip);
+            }
+        }
+
+        private void ValidateSingleIpAddress(string ipAddress)
+        {
+            IPAddress ipVal;
+            if (!IPAddress.TryParse(ipAddress, out ipVal) || ipVal.AddressFamily != System.Net.Sockets.AddressFamily.InterNetwork)
+            {
+                throw new AzPSArgumentException(String.Format("\'{0}\' is not a valid private range ip address", ipAddress), "IpAddress", ErrorKind.UserError);
```

Comment:
It's encouraged that put these exception messages into azure-powershell/src/Network/Network/Properties/Resources.resx and refer them by `Resources.{ErrorName}`
sth like ```throw new AzPSArgumentException(String.Format("Resources.InvalidIpAddress, ipAddress), nameof(IpAddress), ErrorKind.UserError);```
```

Reply:
Updated accordingly